src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Remote/RemoteAutoInsertOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Razor.Remote;
+
+[DataContract]
+internal readonly record struct RemoteAutoInsertOptions
+{
+    [DataMember(Order = 0)]
+    public bool EnableAutoClosingTags { get; init; } = true;
+
+    [DataMember(Order = 1)]
+    public bool FormatOnType { get; init; } = true;
+
+    [DataMember(Order = 2)]
+    public bool InsertSpaces { get; init; } = true;
+
+    [DataMember(Order = 3)]
+    public int TabSize { get; init; } = 4;
```

Comment:
Rather than specify these two properties explicitly, where their values come from a `RazorFormsttingOptions` and their only use is to later construct a `RazorFormattingOptions`, this options type could have a property of type `RazorFormattingOptions` directly. I think it would be clearer, and if we add more formatting options in future, there would be less chance to miss something here, and less having to keep message pack order numbers in line.

Reply:
We are recreating Roslyn FormattingOptions actually, so I'm not sure what the best thing to do here would be. For now I'll just leave the single options here, let's chat on Monday. Happy to do minor cleanup if needed.

Reply:
Yeah, thats my point. So if this type gets `public RazorFormattingOptions FormattingOptions {get; init;} = new()`, then instead of creating a new `RazorFormattingOptions` to pass to the formatting service, it can just pass `options.FormattingOptions`. The `From` method becomes:

```
  public static RemoteAutoInsertOptions From(ClientSettings clientSettings, FormattingOptions formattingOptions)
        => new()
        {
            EnableAutoClosingTags = clientSettings.AdvancedSettings.AutoClosingTags,
            FormatOnType = clientSettings.AdvancedSettings.FormatOnType,
            FormattingOptions = formattingOptions
        };
```

etc etc

Reply:
oops, sorry, missed that you say "Roslyn" formatting options. Still a net win IMO, and can just create a `ToRoslynFormattingOptions` method or extension method, so the code can become `options.FormattingOptions.ToRoslynFormattingOptions()` for example.

Also just realised, another benefit would be that right now we have two spots for a default indent size, and if they get out of whack we could have issues etc.